core/src/main/java/io/grpc/Deadline.java

```diff
@@ -141,4 +141,14 @@ public long timeRemaining(TimeUnit unit) {
   public String toString() {
     return timeRemaining(TimeUnit.NANOSECONDS) + " ns from now";
   }
+
+  @Override
+  public int compareTo(Deadline that) {
+    if (this.deadlineNanos < that.deadlineNanos) {
```

Comment:
The docs only say that "The values returned by this method become meaningful only when the difference between two such values;" it does require the later one to come first. Subtracting in the wrong direction will only change the sign, and will only produce a different but more accurate result for a single nanosecond just past the bound on what the API supports, which is after 292 years.

I'd just `int diff = this.deadlineNanos - that.deadlineNanos`. If still concerned about overflow/underflow, I'd cap deadline creation to be no more than ~100 years in the past or future.